src/components/Gauge.tsx

```diff
@@ -0,0 +1,209 @@
+import { Chart } from 'chart.js';
+import { map } from 'lodash';
+import * as React from 'react';
+import { compose } from 'recompose';
+import styled, { withTheme } from 'styled-components';
+
+import asRendition from '../asRendition';
+import { doughnutDefaults, doughnutLabel } from '../chartPlugins';
+import theme from '../theme';
+
+interface ChartEntry {
+	key?: number;
+	value: number;
+	name: string;
+	color: string;
+}
+
+interface GaugeProps {
+	title: string;
+	placeholderColor?: string;
+	data: ChartEntry[];
+}
+
+const { font } = theme;
+
+const StatusGaugeWrapper = styled.div`
+	display: block;
+	color: ${({ theme }) => theme.colors.text.main};
+
+	.gauge__graph {
+		display: block;
+		width: 100%;
+		max-width: 180px;
+		margin: 0px auto 10px;
+	}
+
+	.gauge__legend {
+		list-style: none;
+		padding: 5px;
+		margin: 16px auto 0;
+		background: white;
+		border-radius: 5px;
+		max-width: 240px;
+		border: 1px solid #ececec;
+		overflow: hidden;
+	}
+
+	.gauge__legend-item {
+		display: flex;
+		padding: 3px 0 3px 2px;
+		line-height: 9px;
+		width: 50%;
+		float: left;
+	}
+
+	.gauge__legend-color {
+		display: inline-block;
+		flex: 0 0 auto;
+		width: 10px;
+		height: 10px;
+		margin-right: 4px;
+	}
+
+	.gauge__legend-name {
+		font-size: 11px;
+	}
+`;
+
+class StatusGauge extends React.Component<GaugeProps, {}> {
+	private DoughnutChart: Chart;
+	private DoughnutCtx: CanvasRenderingContext2D | null;
+
+	componentDidMount() {
+		this.generateDoughnut(this.props);
+	}
+
+	componentWillReceiveProps(nextProps: GaugeProps) {
+		this.generateDoughnut(nextProps);
+	}
+
+	componentWillUnmount() {
+		if (this.DoughnutChart) {
+			this.DoughnutChart.destroy();
```

Comment:
We should probably also set `this.DoughnutChart` to `null` right after `destroy()`, otherwise `destroy()` will be called again if the components re-mounts.
We could actually encapsulate this as a `destroy()` method for this component.